### ISSUE TITLE
"Rename" control for blocks within the Query Loop block are repeated

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#55517]
### Description

The rename option is getting repeated in query loops.

CC @andrewserong 

### Step-by-step reproduction instructions

1. Use the query loop and create a stack within
2. Open the options on the stack

### Screenshots, screen recording, code snippet

<img width="1823" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/283e6fa6-da3f-4acc-85e4-03d1d7796c92">
<details><summary>Block Markup</summary>
<p>

```
<!-- wp:query {"queryId":0,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false},"enhancedPagination":true,"layout":{"type":"constrained"}} -->
<div class="wp-block-query"><!-- wp:post-template -->
<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|default"}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap","justifyContent":"stretch"}} -->
<div class="wp-block-group"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|tight"}},"layout":{"type":"flex","orientation":"vertical"}} -->
<div class="wp-block-group"><!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap"}} -->
<div class="wp-block-group"><!-- wp:post-date {"style":{"elements":{"link":{"color":{"text":"var:preset|color|brand-dark-default"}}}},"textColor":"brand-dark-default","fontSize":"text-sm"} /-->

<!-- wp:post-author-name {"style":{"elements":{"link":{"color":{"text":"var:preset|color|brand-dark-default"}}}},"textColor":"brand-dark-default","fontSize":"text-sm"} /--></div>
<!-- /wp:group -->

<!-- wp:post-title /--></div>
<!-- /wp:group -->

<!-- wp:post-excerpt {"moreText":"\u003cem\u003eRead more\u003c/em\u003e","showMoreOnNewLine":false,"style":{"elements":{"link":{"color":{"text":"var:preset|color|dark-lighter"}}}}} /-->

<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|close","bottom":"var:preset|spacing|close","left":"var:preset|spacing|close","right":"var:preset|spacing|close"}}},"backgroundColor":"dark-contrast","layout":{"type":"flex","flexWrap":"nowrap"},"fontSize":"text-sm"} -->
<div class="wp-block-group has-dark-contrast-background-color has-background has-text-sm-font-size" style="padding-top:var(--wp--preset--spacing--close);padding-right:var(--wp--preset--spacing--close);padding-bottom:var(--wp--preset--spacing--close);padding-left:var(--wp--preset--spacing--close)"><!-- wp:paragraph -->
<p>Tags:</p>
<!-- /wp:paragraph -->

<!-- wp:post-terms {"term":"post_tag","style":{"elements":{"link":{"color":{"text":"var:preset|color|brand-dark-default"}}}}} /--></div>
<!-- /wp:group --></div>
<!-- /wp:group -->
<!-- /wp:post-template -->

<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:query-pagination {"layout":{"type":"flex","justifyContent":"space-between"}} -->
<!-- wp:query-pagination-previous /-->

<!-- wp:query-pagination-next /-->
<!-- /wp:query-pagination --></div>
<!-- /wp:group -->

<!-- wp:query-no-results -->
<!-- wp:paragraph {"placeholder":"Add text or blocks that will display when a query returns no results."} -->
<p></p>
<!-- /wp:paragraph -->
<!-- /wp:query-no-results --></div>
<!-- /wp:query -->
```
</p>
</details> 

### Environment info

- WP 6.3.2
- GB 16.8.1
- Custom theme

### Please confirm that you have searched existing issues in the repo.

No

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by richtabor** [comment]
I found this today as well. Perhaps we can get this in as a bug fix for 6.4 @annezazu @getdave @mikachan.

**Comment by annezazu** [comment]
Agreed. Adding this to to do as this is a gnarly bug.

**Comment by annezazu** [comment]
Had more time to loop back on this and wanted to note that I can replicate with 6.4-RC1 (just to confirm that it is 6.4 related).

**Comment by richtabor** [comment]
Yea, "Rename" in the menu is new to 6.4 if I'm remembering correctly.

**Comment by getdave** [comment]
100% agreed this needs to be fixed for 6.4. Probably via https://github.com/WordPress/gutenberg/issues/55338.

**Comment by getdave** [comment]
I realised that any fix here will need to be against the 6.4 branch and _not_ `trunk` because 6.4 will not have the renaming for _all_ blocks (only Group blocks) that is present in `trunk`.

**Comment by getdave** [comment]
Speaking to a few folks away from Github we determined the fix will require a refactoring the approach away from hooks. There is a PR WIP.
